### PR TITLE
Fix race condition in bicep / ARM template

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
Without this change, virtually all deployments will fail when deploying some of the extensions (MSDeploy or App Insights) for our services.

### Description
- Make each service has its own virtualNetworkConnections resource
- Add dependsOn clause for the service extensions on their respective virtualNetworkConnections resource
- Remove some crud that had accumulated in the template files

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations

### CODE REVIEW

File: scripts/deploy/main.bicep [hunk 1]
```diff
@@ -463,7 +464,6 @@ resource appServiceMemoryPipelineConfig 'Microsoft.Web/sites/config@2022-09-01'
     minTlsVersion: '1.2'
     netFrameworkVersion: 'v6.0'
     use32BitWorkerProcess: false
-    vnetName: webSubnetConnection.name
```
Comment (glahaye) [inline scripts/deploy/main.bicep:466]: Not needed

File: scripts/deploy/main.bicep [hunk 2]
```diff
@@ -883,16 +884,6 @@ resource virtualNetwork 'Microsoft.Network/virtualNetworks@2021-05-01' = {
           privateLinkServiceNetworkPolicies: 'Enabled'
         }
       }
-      {
```
Comment (glahaye) [inline scripts/deploy/main.bicep:886]: Should have been removed along with the rest of the PostGres stuff

File: scripts/deploy/object.txt [hunk 3]
```diff
@@ -1,94 +0,0 @@
-{
```
Comment (glahaye) [inline scripts/deploy/object.txt:1]: Never should have been checked in in the first place

**Review by teresaqhoang — APPROVED** [review]

**Review by alliscode — APPROVED** [review]

**Comment by cricex** [comment]
I see this PR has been merged, but I am still getting the same race error using ARM "Deploy to Azure" option. I am pointing to:

- https://aka.ms/copilotchat/webapi/latest
- https://aka.ms/copilotchat/memorypipeline/latest
- https://aka.ms/copilotchat/websearcher/latest

Are these URIs updated w/ the latest fix?

**Pull request merged**